### PR TITLE
Fix logentry.channel json marshaling/unmarshaling

### PR DESCRIPTION
**Pull request by StupidScience**
Hello.

I've faced an issue when unmarshalling and marshalling back of LogEntry lead to results that differ from initial.

First of all just because of absent json tags, so it lead to `"Type"` instead of `"type"`, so the next unmarshaling was broken.
Also "fixing" tags leads to nested object and
```json
{
    ...
    "channel": {
	    "type": "web_trigger",
	    "summary": "My new incident",
	    "details_omitted": false
    }
}
```

transforms to
```json
{
    ...
    "channel": {
	    "type": "web_trigger",
            "raw": {
	        "summary": "My new incident",
	        "details_omitted": false
            }
    }
}
```

So this PR will fix this, however it require Channel type changes.

Looking forward for your feedback.

### CODE REVIEW

**Comment by theckman** [comment]
So this change is technically breaking API compatibility, and under normal circumstances would be a candidate for a v2.0.0 release. That said, digging further I'm not confident that the old format was actually functional based on the API documentation. As such, I'm not confident we should hold this to the same compatibility guarantee under the v1.0.0 released.

I'm going to sync up with @stmcallister to propose a `v1.5.0` release that includes this change, and other "breaking" changes, which fix an implementation that has never worked reliably.

**Comment by StupidScience** [comment]
@theckman may be better instead of breaking API would be to provide custom `MarshalJSON()` function? It won't break things but should resolve issue as well. WDYT?

**Comment by theckman** [comment]
Yeah, that may be more ideal. Unmarshal into a map, extract the type field out, and then the rest of the map on the Raw field.

**Review by theckman — CHANGES_REQUESTED** [review]
Should have made that comment (^) while requesting changes.

File: log_entry.go
Comment (theckman) [inline log_entry.go:118]: Sorry, I'm going to be a real pain in your rear here... For some reason it didn't dawn on me that removing this would also be a breaking change, and also make it not the raw object anymore. Would you have the spare moment to squash out this line? If not I can fix it after merging this PR.

Thank you so much for this contribution.

File: log_entry.go
Comment (theckman) [inline log_entry.go:137]: I suspect `json.Marshal()` effectively does the same thing. How would you feel about simplifying this error handling block all down to `return json.Marshal(raw)`?

**Review by theckman — CHANGES_REQUESTED** [review]
Requesting two final adjustments, one of them is my fault. Afterwards I think this will be :+1:

File: log_entry.go
Comment (StupidScience) [reply log_entry.go:137]: Nice catch. I'll fix it

File: log_entry.go
Comment (StupidScience) [reply log_entry.go:118]: Not a problem at all. I'll send update shortly

**Review by theckman — APPROVED** [review]

**Pull request merged**
